src/main/java/io/apicurio/datamodels/asyncapi/models/AaiOperationBase.java

```diff
@@ -47,12 +53,40 @@ public AaiOperationBase(Node parent, String opType) {
         }
     }
 
+    /**
+     * Constructor.
+     * @param parent
+     */
     public AaiOperationBase(Node parent) {
         this(parent, null);
     }
 
+    /**
+     * Adds a tag.
+     * @param tag
+     */
+    public void addTag(AaiTag tag) {
+        if(tags == null)
+            tags = new LinkedList<>();
+        tags.add(tag);
+    }
 
-    public abstract List<AaiProtocolInfo> getProtocolInfoList();
-    public abstract void addTag(AaiTag tag);
-    public abstract void addProtocolInfo(AaiProtocolInfo value);
+    /**
+     * Adds a tag.
+     *
+     * @param name
+     * @param description
+     */
+    public AaiTag addTag(String name, String description) {
```

Comment:
I think I made the methods in Aai* classes abstract to keep the implementation specific for each version (no need for createTag()), but this also works

Reply:
Yeah I think the methods should be abstract when there is an expectation of differentiation.  But for things like `add` and `get` methods, I do not think that the different versions of AAI models would require different logic.  So I kept all of the `create` methods abstract (or tried to - I may have missed some?) but moved the others into the base.  We may need to go through all the model classes to ensure that approach is consistent.